### PR TITLE
WIP: Resolve coincident topology in _Brain

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR fixes the part of https://github.com/mne-tools/mne-python/issues/7599 about overlapping surfaces.

### CODE REVIEW

File: mne/viz/_brain/_brain.py [hunk 1]
```diff
@@ -474,6 +474,10 @@ def add_data(self, array, fmin=None, fmid=None, fmax=None,
                 actor, mesh = mesh_data
                 # add metadata to the mesh for picking
                 mesh._hemi = hemi
+                mapper = actor.GetMapper()
+                mapper.SetResolveCoincidentTopologyToPolygonOffset()
+                mapper.SetRelativeCoincidentTopologyPolygonOffsetParameters(
+                    -1., -1.)
```
Comment (larsoner) [inline mne/viz/_brain/_brain.py:480]: What I would do here is make it implicitly that the order of addition sets the zorder (things plotted later end up on top). So `brain` should have something like a `self._polygon_offset_counter = 0` in init and anything you add afterward does `self._polygon_offset_counter += 1` and then `mapper.SetRelativeCoincidentTopologyPolygonOffsetParameters(-self._polygon_offset_counter, -self._polygon_offset_counter)` or so. WDYT?

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_brain.py:480]: From my experience with polygon offset, especially if many surfaces overlap, some artifacts can appear. Also, I checked the doc of SetRelativeCoincidentTopologyPolygonOffsetParameter and it accepts double so maybe we have some room for tweaks here.

Ref: https://vtk.org/doc/nightly/html/classvtkMapper.html#ad49f0d5f3086b3a79f542cb9e801d8ac

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_brain.py:480]: I'll go with `self._polygon_offset_counter` and modify accordingly if we run into artifacts :+1:

Comment (larsoner) [reply mne/viz/_brain/_brain.py:480]: It seems like as soon as we do a `brain.add_annotation` plus a `brain.add_data` we will have a problem, but I guess we can deal with it then

**Pull request merged**